### PR TITLE
Topic Proxy: terminate orphaned proxy sessions upon cluster rehashing.

### PR DESCRIPTION
**Pull request by aforge**

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -1210,6 +1210,43 @@ func (c *Cluster) invalidateProxySubs() {
...
+	}
+	globals.hub.topics.Range(func(_, v interface{}) bool {
+		topic := v.(*Topic)
+		if topic.isProxy {
+			// Topic is a proxy. Continue.
+			return true
+		}
+		for s := range topic.sessions {
+			if s.isProxy() {
+				_, originActive := activeNodeMap[s.clnode.name]
```
Comment (or-else) [inline server/cluster.go:1229]: More idiomatic
`if _, originActive := activeNodeMap[s.clnode.name]; !originActive { ...`

File: server/cluster.go [hunk 2]
```diff
@@ -1210,6 +1210,43 @@ func (c *Cluster) invalidateProxySubs() {
...
+		return true
+	})
+	// Unsubscribe orphaned sessions from their master topics.
+	// Stop orphaned sessions.
+	for s, t := range sessions {
+		leave := &sessionLeave{
+			terminateProxyConnection: true,
+			sess:                     s,
+		}
+		t.unreg <- leave
```
Comment (or-else) [inline server/cluster.go:1245]: No need for variable allocation:
```
t.unreg <- &sessionLeave{
			terminateProxyConnection: true,
			sess:                     s,
		}
```
Only if you like it for readability.

File: server/cluster.go (hunk 1 above)
Comment (aforge) [reply server/cluster.go]: Right. Changed.

File: server/cluster.go (hunk 2 above)
Comment (aforge) [reply server/cluster.go]: Done.

**Pull request merged**